svelte/src/scripts/draw/helper/tree-layout.ts

```diff
@@ -0,0 +1,142 @@
+// PLAN B CHARGE FORCE!
+
+import * as d3 from "d3";
+import type { simluationLinkType, simluationNodeDatumType } from "./types";
+
+interface simluationNodeDatumTypeExt extends simluationNodeDatumType {
+	treeData?: {
+		targets: Map<simluationNodeDatumType, number>; // Store nodes at lower level
+		level: number; // The depth of the node in the tree
+	}
+};
+
+/** 
+ * Caculates a set of arboricities on the given graph, but only on the top level in the dependency hierarchy.
+ * (Not necicarilly maximal or efficient)
+ * 
+ * Stores data in a new property treeData on the node (in place)
+ * 
+ * TODO move to parser
+ * */ 
+function findArboricity(nodes: simluationNodeDatumTypeExt[]) {
+	// Helper functions
+	function getOutLinks(node: simluationNodeDatumTypeExt): simluationLinkType[] {
+		return node.members.flatMap(m => getOutLinks(m))
+		.concat(node.outgoingLinks ?? []);
+	};
+
+	function getCompoundNode(node: simluationNodeDatumType): simluationNodeDatumType {
```

Comment:
uuh sorry stupid question here, what is this function for?

Reply:
It gets the ansector of a node which is actually in the simulation. Will probably be moved to a helper function eventually.